homeassistant/components/calendar/__init__.py

```diff
@@ -35,6 +52,19 @@
 # mypy: disallow-any-generics
 
 
+CALENDAR_EVENT_SCHEMA = vol.Schema(
+    {
+        vol.Optional(EVENT_UID): cv.string,
+        vol.Optional(EVENT_RECURRENCE_ID): cv.string,
+        vol.Optional(EVENT_START): vol.Any(cv.date, cv.datetime),
+        vol.Optional(EVENT_END): vol.Any(cv.date, cv.datetime),
+        vol.Optional(EVENT_SUMMARY): cv.string,
+        vol.Optional(EVENT_DESCRIPTION): cv.string,
+        vol.Optional(EVENT_RRULE): cv.string,
```

Comment:
The RRULE spec is defined here:
https://www.rfc-editor.org/rfc/rfc5545#section-3.3.10

The rrule value is currently validated in the integration library here: https://github.com/allenporter/ical/blob/main/ical/types/recur.py

However, what we can also do is add more validation here of the form:
- verifying that it parses as an a rrule at all
- verifying specific parts of rrule home assistant wants to support and enforce across all integrations

I think there are some parts of rrule we don't want to or need to support (e.g. recurrence rules that repeat more often than daily), so we could enforce that here rather than in the integration.  I think validating the `FREQ` field could be a reasonable thing to do here, but maybe not enforcing constraints on all the other rule parts.

If we ever support event editing, one challenge is that you might imagine that an integration could support an rrule that the frontend doesn't know how to create. We can either pass those through unvalidated, but probably could be simplest to just not support updates.

So I lean towards using `datetutil.rrule` for a "light" validation of `FREQ` here.  Seem OK?

Reply:
Sounds good.

Reply:
Added.